api/doc/action-audit-queries.md

```diff
@@ -0,0 +1,212 @@
+# Action Audit Queries
+
+## Schema
+The Action Audit schema is a flattened schema where each row of the database represents
+an action taken by a user or a user-visisble action taken on their behalf by either an
+administrator, another standard user, or the system itself.
+
+### Actions and Events
+A user action may be recorded as multiple rows in the database. This happens because we have
+a flat, denormalized schema and can't represent composite structures in a single row. In order to
+keep related events in an action together, a randomly generated Action ID is used. This essentially
+acts as a paperclip to keep events from the same action together.
+
+### Agent Types
+There are three agent types currently:
+- USER: standard AoU users. 
+- ADMINISTRATOR: Users with Admin privileges. Used for BYPASS events and the like
+- SYSTEM: Some part of the All of Us software stack. Typically used for automated events,
+or events that happen as a consequence of some other event.
+
+### Target Types
+A Target is a thing being operated on during an Action. The following target types are largely
+self-explanatory. The design document has additional details.
+ - USER
+ - WORKBENCH
+ - PROFILE
+ - ACCOUNT
+ - NOTEBOOK
+ - NOTEBOOK_SERVER
+ - DATASET
+ - CONCEPT_SET
+ - COHORT
+ - CREDIT
+ - WORKSPACE
+
+## Target Properties
+The target properties are key-value string pairs and are not in a strictly
+controlled schema, but are rather enforced by application constraints. Each target type 
+supports one or more properties, and these are outlined below:
+
+| Target Type | Property  |
+|-------------|----------|
+| WORKSPACE   | name     |
+| WORKSPACE | intended_study |
+| WORKSPACE | creator | 
+| WORKSPACE | additional_notes | 
+| WORKSPACE | anticipated_findings | 
+| WORKSPACE | disease_of_focus | 
+| WORKSPACE | reason_for_all_of_us | 
+| WORKSPACE | namespace | 
+| WORKSPACE | cdr_version_id | 
+| PROFILE | user_name |
+| PROFILE | contact_email |
+| PROFILE | data_access_level |
+| PROFILE | given_name |
+| PROFILE | phone_number |
+| PROFILE | current_position |
+| PROFILE | organization |
+| PROFILE | disabled |
+| PROFILE | about_you |
+| PROFILE | area_of_research |
+| PROFILE | institutional_affiliations |
+| PROFILE | demographic_survey |
+| PROFILE | address |
+| ACL | ACCESS_LEVEL |
```

Comment:
It would be helpful to list all column names and corresponding datatype and descriptions for workbench_action_audit table. Or maybe link the design doc if that info is included?

Reply:
Good point. I kinda rushed this so I could get the queries in. I'll add some description here.

This doc is public, but the design doc is private, so I can't really link from here.